app/models/concerns/hostext/search.rb

```diff
@@ -226,6 +226,7 @@ def search_cast_facts(key, operator, value)
       def search_by_os_major(key, operator, value)
         condition = sanitize_sql_for_conditions(["CAST(major AS DECIMAL) #{operator} ?", value_to_sql(operator, value.to_f)])
         operatingsystem_ids = Operatingsystem.select(:id).where(condition).pluck('operatingsystems.id').join(',')
+        operatingsystem_ids = ['-1'] if operatingsystem_ids.empty?
```

Comment:
I think this shouldn't be an array here (since it is joined first):
```suggestion
        operatingsystem_ids = '-1' if operatingsystem_ids.empty?
```

Reply:
done @tbrisker. thanks